core/src/main/scalajvm/scalapb/zio_grpc/server/ZServerCallHandler.scala

```diff
@@ -107,56 +109,23 @@ object ZServerCallHandler {
       call: ZServerCall[Res],
       stream: ZStream[Any, StatusException, Res]
   ): ZIO[Any, StatusException, Unit] = {
-    def takeFromQueue(
-        queue: Dequeue[Exit[Option[StatusException], Res]]
-    ): ZIO[Any, StatusException, Unit] =
-      queue.takeAll.flatMap(takeFromCache(_, queue))
+    val backpressureSink = {
+      def go: ZChannel[Any, ZNothing, Chunk[Res], Any, StatusException, Chunk[Res], Unit] =
+        ZChannel.readWithCause(
+          (x: Chunk[Res]) =>
+            ZChannel.succeed {
+              x.foreach(call.call.sendMessage)
+            } *> ZChannel.suspend(if (call.call.isReady()) go else ZChannel.fromZIO(call.awaitReady) *> go),
```

Comment:
Would this be more appropriate? Since `call.call.sendMessage` can fail with `IllegalStateException` if headers are not sent or call is closed ?

```scala
      def go: ZChannel[Any, ZNothing, Chunk[Res], Any, Throwable, Chunk[Res], Unit] =
        ZChannel.readWithCause(
          (ls: Chunk[Res]) =>
            ZChannel.fromZIO(ZIO.attempt(ls.foreach(call.call.sendMessage))) *>
              ZChannel.suspend(if (call.call.isReady) go else ZChannel.fromZIO(call.awaitReady) *> go),
          (cause: Cause[Throwable]) => ZChannel.failCause(cause),
          _ => ZChannel.unit
        )

      ZSink.fromChannel(go).mapError { e =>
        Status.INTERNAL.withDescription(e.getMessage).withCause(e).asException
      }
```

Tried it and I saw no performance degradation while running the streaming request.

Reply:
Nice, done.